### ISSUE TITLE
[HoltWinters] Weekly data and 52 season_length (1 year) not working

### ISSUE THREAD

**Issue by nelsoncardenas** [Nixtla/statsforecast#480]
### What happened + What you expected to happen

1. Bug

I have 3 years data, and I fit my training data with week granularity, so I need to put seasonality equals to `52` (for the whole year).
```python
hw_model = HoltWinters(season_length=52, error_type="A").fit(
    train_data["sales"].to_numpy()
)
forecast = hw_model.predict(h=2)
```

But I got this result:

```json
{'mean': array([nan, nan])}
```

2. expected behavior

Some output with float numbers, for instance:

```python
{'mean': array([65.12, 65.75])}
```

### Versions / Dependencies

python 3.9.5
Code in Databricks.
```python
numpy==1.24.3
pandas==1.5.3
statsforecast==1.5.0
prophet==1.1.2
```

### Reproduction script

```python
import numpy as np
import pandas as pd
from statsforecast.models import HoltWinters
import matplotlib.pyplot as plt

# Generate simulated sales data for 3 years (156 weeks)
np.random.seed(123)
sales = 0.5 + np.random.randn(156)
sales = np.cumsum(sales)

# Create a pandas DataFrame with the sales data
dates = pd.date_range(start="2020-01-01", periods=len(sales), freq="W")
df = pd.DataFrame({"sales": sales}, index=dates)

# Split the data into training and testing sets
train_size = int(0.8 * len(df))
train_data = df.iloc[:train_size]
test_data = df.iloc[train_size:]

# Train a Holt-Winters model using the training data
hw_model = HoltWinters(season_length=52, error_type="A").fit(
    train_data["sales"].to_numpy()
)
# Use the model to make predictions for the next two weeks
forecast = hw_model.predict(h=2)
print(forecast)
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by nelsoncardenas** [comment]
Update:

If I use the Z parameter the predictions beging to work:
```python
hw_model = HoltWinters(season_length=52, error_type="Z").fit(
    train_data["sales"].to_numpy()
)
```
`
{'mean': array([64.47814443, 65.00342111])}`

**Comment by nelsoncardenas** [comment]
To be honest I'm really confused about the cause of the error. My last update was just for the specific dummy data I shared, but if I add seasonality and trend to the data, the error is the same event when using `error_type_="Z"`.

### Reproduction script

```python
import numpy as np
import pandas as pd
from statsforecast.models import HoltWinters
import matplotlib.pyplot as plt

# Generate simulated sales data for 3 years (156 weeks)
np.random.seed(123)
sales = 10 + np.random.randn(156) + 0.05 * np.arange(156)
#sales = np.cumsum(sales)

# Add seasonality pattern for July
for i in range(3):
    sales[i*52 + 48] += 20 + i * 15
    sales[i*52 + 47] += 5 + i * 6.15

# Create a pandas DataFrame with the sales data
dates = pd.date_range(start="2020-01-01", periods=len(sales), freq="W")
df = pd.DataFrame({"sales": sales}, index=dates)

# Split the data into training and testing sets
train_size = int(0.9 * len(df))
train_data = df.iloc[:train_size]
test_data = df.iloc[train_size:]

# Train a Holt-Winters model using the training data
hw_model = HoltWinters(season_length=52, error_type="Z").fit(
    train_data["sales"].to_numpy()
)
# Use the model to make predictions for the next two weeks
forecast = hw_model.predict(h=len(test_data))["mean"]
print(forecast)


# Plot the actual sales data and the predicted sales data
plt.figure(figsize=(10, 6))
plt.plot(train_data.index, train_data["sales"], label="Training data")
plt.plot(test_data.index, test_data["sales"], label="Testing data")
plt.plot(test_data.index, forecast, label="Forecast")
plt.legend()
plt.show()
```

**Comment by FedericoGarza** [comment]
hey @nelsoncardenas! Thank you very much for reporting this in a well-documented issue. It was solved in #483. I tested installing the library from the main and now it works as expected:

<img width="776" alt="image" src="https://github.com/Nixtla/statsforecast/assets/10517170/8cafff30-11d1-4944-b1d5-f1e3a1867baa">

We will release a new version to pip soon. :)

Feel free to reopen the issue if it persists.

**Comment by nelsoncardenas** [comment]
Thank you so much @FedericoGarza , I thought this issue would take at least a month 😅, happy to see this working!!!
